### PR TITLE
Migrate GeoJSONUtilsTest to assertj

### PR DESCRIPTION
**Pull request by mkleen**
## Checklist

 - [x] Added an entry in the latest `docs/appendices/release-notes/<x.y.0>.rst` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in the latest `docs/appendices/release-notes/<x.y.0>.rst`
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java [hunk 1]
```diff
@@ -149,97 +146,106 @@ public void testMap2Shape() throws Exception {
...
+        assertThat(shape).isExactlyInstanceOf(JtsGeometry.class);
+        assertThat(((JtsGeometry) shape).getGeom()).isExactlyInstanceOf(LineString.class);
 
     }
 
     @Test
     public void testInvalidMap() throws Exception {
-        assertThrows(IllegalArgumentException.class,
-                     () -> GeoJSONUtils.map2Shape(Map.<String, Object>of()),
-                     "Cannot convert Map \"{}\" to shape");
```
Comment (mkleen) [inline server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java:161]: This argument was not the exception message, but the assert message. Therefore it is changed now.

File: server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java [hunk 2]
```diff
@@ -113,9 +109,9 @@ public void testPoint2Map() throws Exception {
         Point point = GEOMETRY_FACTORY.createPoint(new Coordinate(0.0, 0.0));
         Shape shape = new JtsPoint(point, JtsSpatialContext.GEO);
         Map<String, Object> map = GeoJSONUtils.shape2Map(shape);
-        assertThat(map, hasEntry("type", (Object) "Point"));
-        assertThat(map.get("coordinates").getClass().isArray(), is(true));
-        assertThat(((double[]) map.get("coordinates")).length, is(2));
+        assertThat(map).containsEntry("type", "Point");
+        assertThat(map.get("coordinates").getClass().isArray()).isTrue();
+        assertThat(((double[]) map.get("coordinates")).length).isEqualTo(2);
```
Comment (matriv) [inline server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java:114]: very minor, you can use:
```
assertThat((double[]) map.get("coordinates")).hasSize(2);
```

File: server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java [hunk 3]
```diff
@@ -125,22 +121,23 @@ public void testMapFromWktRoundTrip() throws Exception {
...
-                         "multilinestring (((10.05  10.28  3.4  8.4, 20.95  20.89  4.5  9.5),\n" +
-                         " \n" +
-                         "( 20.95  20.89  4.5  9.5, 31.92  21.45  3.6  8.6)))"),
-                     "Cannot convert WKT \"multilinestring (((10.05  10.28  3.4  8.4, 20.95  20.89  4.5  9.5)");
+        assertThatThrownBy(
+            () -> GeoJSONUtils.wkt2Map(
+                "multilinestring (((10.05  10.28  3.4  8.4, 20.95  20.89  4.5  9.5),\n" +
+                    " \n" +
+                    "( 20.95  20.89  4.5  9.5, 31.92  21.45  3.6  8.6)))"),
+            "Cannot convert WKT \"multilinestring (((10.05  10.28  3.4  8.4, 20.95  20.89  4.5  9.5)")
```
Comment (matriv) [inline server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java:139]: shouldn't this be `.hasMessageStartingWith("Cannot convert WKT \"multilinestring (((10.05  10.28  3.4  8.4, 20.95  20.89  4.5  9.5)");`?

File: server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java [hunk 4]
```diff
@@ -125,22 +121,23 @@ public void testMapFromWktRoundTrip() throws Exception {
...
+            .isInstanceOfAny(IllegalArgumentException.class);
```
Comment (matriv) [inline server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java:140]: here and elsewhere, why not: `.isExactlyInstanceOf` ?

File: server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java (hunk 1 above)
Comment (matriv) [reply server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java:161]: then please add:
`.hasMessage("Cannot convert GeoJSON \"{}\" to shape")`.

File: server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java [hunk 5]
```diff
@@ -149,97 +146,106 @@ public void testMap2Shape() throws Exception {
...
 
     }
 
     @Test
     public void testInvalidMap() throws Exception {
-        assertThrows(IllegalArgumentException.class,
-                     () -> GeoJSONUtils.map2Shape(Map.<String, Object>of()),
-                     "Cannot convert Map \"{}\" to shape");
+        assertThatThrownBy(
+            () -> GeoJSONUtils.map2Shape(Map.<String, Object>of()))
```
Comment (matriv) [inline server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java:157]: very minor:
```suggestion
            () -> GeoJSONUtils.map2Shape(Map.of()))
```

**Review by matriv — APPROVED** [review]
Thx! left some comments though.

File: server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java (hunk 1 above)
Comment (matriv) [reply server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java:161]: apologies, ghub diff hid this for me.

**Pull request merged**